backend/src/schema/resolvers/reports.js

```diff
@@ -11,12 +11,31 @@ export default {
         description: description,
       }
 
+      const reportQueryRes = await session.run(
+        `
+        match (u:User {id:$submitterId}) -[:REPORTED]->(report)-[:REPORTED]-> (resource {id: $resourceId}) 
+        return  labels(resource)[0] as label
+        `,
+        {
+          resourceId: id,
+          submitterId: user.id,
+        },
+      )
+      const [rep] = reportQueryRes.records.map(record => {
+        return {
+          label: record.get('label'),
+        }
+      })
+
+      if (rep) {
+        throw new Error(rep.label)
+      }
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-21T12:46:34Z" title="Friday, June 21st 2019, 2:46:34 pm +02:00">Jun 21, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -11,12 +11,32 @@ export default {
         description: description,
       }
 
+      const reportQueryRes = await session.run(
+        `
+        match (u:User {id:$submitterId}) -[:REPORTED]->(report)-[:REPORTED]-> (resource {id: $resourceId}) 
+        return  report, labels(resource)[0] as type
+        `,
+        {
+          resourceId: id,
+          submitterId: user.id,
+        },
+      )
+      const [rep] = reportQueryRes.records.map(record => {
+        return {
+          report: record.get('report'),
+          type: record.get('type'),
+        }
+      })
+
+      if (rep) {
+        throw new Error(rep.type)
```


Why do you throw an error here? Wouldn't it be enough to `return rep`?

Reply:
> [<img alt="ogerly" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-06-21T14:01:02Z" title="Friday, June 21st 2019, 4:01:02 pm +02:00">Jun 21, 2019</time>_

---


the error is output if a report already exists. 
throw is as I understood it equal to return. 
rep.type is the type of the report (user,post or comment). 
new Error only outputs the string . no additional information that would otherwise come when an error is reported from the database. 
in the frontend the respective error message is displayed. 
throw aborts the code like return.